WordPress/src/main/java/org/wordpress/android/ui/activitylog/RewindProgressChecker.kt

```diff
@@ -1,63 +1,51 @@
 package org.wordpress.android.ui.activitylog
 
-import android.os.Handler
-import android.os.Looper
-import org.wordpress.android.fluxc.Dispatcher
-import org.wordpress.android.fluxc.generated.ActivityLogActionBuilder
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.runBlocking
+import org.wordpress.android.fluxc.action.ActivityLogAction.FETCH_REWIND_STATE
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FINISHED
 import org.wordpress.android.fluxc.store.ActivityLogStore
 import org.wordpress.android.fluxc.store.ActivityLogStore.FetchRewindStatePayload
-import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.OnRewindStatusFetched
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlin.coroutines.experimental.coroutineContext
 
 @Singleton
 class RewindProgressChecker
-@Inject
-constructor(val activityLogStore: ActivityLogStore, val siteStore: SiteStore, val dispatcher: Dispatcher) {
-    private var handler: Handler = Handler(Looper.getMainLooper())
-    private var checkState: Runnable? = null
-    var isRunning: Boolean = false
-
+@Inject constructor(private val activityLogStore: ActivityLogStore) {
     companion object {
         const val CHECK_DELAY_MILLIS = 10000L
     }
 
-    fun startNow(site: SiteModel, restoreId: Long) {
-        start(site, restoreId, true)
+    suspend fun startNow(site: SiteModel, restoreId: Long): OnRewindStatusFetched? {
+        return start(site, restoreId, true)
     }
 
-    fun start(site: SiteModel, restoreId: Long, now: Boolean = false) {
-        isRunning = true
-        checkState = object : Runnable {
-            override fun run() {
-                val rewindStatusForSite = activityLogStore.getRewindStatusForSite(site)
-                val rewind = rewindStatusForSite?.rewind
-                rewind?.let {
-                    if (rewind.status == FINISHED && rewind.restoreId == restoreId) {
-                        isRunning = false
-                        return
-                    }
-                }
-
-                val action = ActivityLogActionBuilder.newFetchRewindStateAction(FetchRewindStatePayload(site))
-                dispatcher.dispatch(action)
-
-                handler.postDelayed(this, CHECK_DELAY_MILLIS)
-            }
-        }
-        if (now) {
-            handler.post(checkState)
-        } else {
-            handler.postDelayed(checkState, CHECK_DELAY_MILLIS)
+    suspend fun start(
+        site: SiteModel,
+        restoreId: Long,
+        now: Boolean = false,
+        checkDelay: Long = CHECK_DELAY_MILLIS
+    ) = runBlocking(coroutineContext) {
```

Comment:
Do we really want to block the current thread? This method is called [from a coroutine ](https://github.com/wordpress-mobile/WordPress-Android/pull/8264/files#diff-3615dedbb485172c51a2a8af3ac0d42aR118), but the docs mention that [they shouldn't be](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.experimental/run-blocking.html).

Reply:
great point! I'll check it